nifi-nar-bundles/nifi-solr-bundle/nifi-solr-processors/src/main/java/org/apache/nifi/processors/solr/FetchSolr.java

```diff
@@ -0,0 +1,401 @@
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import static org.apache.nifi.processors.solr.SolrUtils.SOLR_TYPE;
+import static org.apache.nifi.processors.solr.SolrUtils.COLLECTION;
+import static org.apache.nifi.processors.solr.SolrUtils.JAAS_CLIENT_APP_NAME;
+import static org.apache.nifi.processors.solr.SolrUtils.SSL_CONTEXT_SERVICE;
+import static org.apache.nifi.processors.solr.SolrUtils.SOLR_SOCKET_TIMEOUT;
+import static org.apache.nifi.processors.solr.SolrUtils.SOLR_CONNECTION_TIMEOUT;
+import static org.apache.nifi.processors.solr.SolrUtils.SOLR_MAX_CONNECTIONS;
+import static org.apache.nifi.processors.solr.SolrUtils.SOLR_MAX_CONNECTIONS_PER_HOST;
+import static org.apache.nifi.processors.solr.SolrUtils.ZK_CLIENT_TIMEOUT;
+import static org.apache.nifi.processors.solr.SolrUtils.ZK_CONNECTION_TIMEOUT;
+import static org.apache.nifi.processors.solr.SolrUtils.SOLR_LOCATION;
+import static org.apache.nifi.processors.solr.SolrUtils.BASIC_USERNAME;
+import static org.apache.nifi.processors.solr.SolrUtils.BASIC_PASSWORD;
+import static org.apache.nifi.processors.solr.SolrUtils.RECORD_WRITER;
+
+@Tags({"Apache", "Solr", "Get", "Query", "Records"})
+@InputRequirement(InputRequirement.Requirement.INPUT_ALLOWED)
+@CapabilityDescription("Queries Solr and outputs the results as a FlowFile in the format of XML or using a Record Writer")
+@WritesAttributes({
+        @WritesAttribute(attribute = "solr.cursor.mark", description = "Cursor mark can be used for scrolling Solr"),
+        @WritesAttribute(attribute = "solr.status.code", description = "Status code of Solr request. A status code of 0 indicates that the request was successfully processed"),
+        @WritesAttribute(attribute = "solr.query.time", description = "The elapsed time to process the query (in ms)"),
+        @WritesAttribute(attribute = "mime.type", description = "The mime type of the data format"),
+        @WritesAttribute(attribute = "fetchsolr.exeption.class", description = "The Java exception class raised when the processor fails"),
+        @WritesAttribute(attribute = "fetchsolr.exeption.message", description = "The Java exception message raised when the processor fails")
+})
+public class FetchSolr extends SolrProcessor {
+
+    public static final AllowableValue MODE_XML = new AllowableValue("XML");
+    public static final AllowableValue MODE_REC = new AllowableValue("Records");
+    public static final String MIME_TYPE_JSON = "application/json";
+    public static final String ATTRIBUTE_CURSOR_MARK = "solr.cursor.mark";
+    public static final String ATTRIBUTE_SOLR_STATUS = "solr.status.code";
+    public static final String ATTRIBUTE_QUERY_TIME = "solr.query.time";
+    public static final String EXCEPTION = "fetchsolr.exeption";
+    public static final String EXCEPTION_MESSAGE = "fetchsolr.exeption.message";
+
+    public static final PropertyDescriptor SOLR_QUERY_STRING = new PropertyDescriptor

```

Comment:
Are you going to support parameters like fq? If you intend to put them here, you might want to consider break it out into some additional fields for the sake of readability and ease of use.

Reply:
For parameters such as fq a single designated property will not be sufficient, as it is possible (and sometimes necessary) to define multiple filter queries. Furthermore, depending on the query parser, there are various parameters that can be used for queries. I considered to make the most common query parameters configurable via designated properties, and to provide the option of additional parameters via dynamic properties. However, I personally expect this single field to be more straightforward. I expect the majority of the users to test queries in Solr directly and to paste the query string afterwards into the query property. Which option do you prefer?

Reply:
I think that's fine. I did something similar in a PR I have open for ElasticSearch + Kibana. If that's how you see most users doing it, seems fine.

Reply:
I personally prefer "Solr Query String" to mean a lucene style query like "(foo AND bar)" and then have properties for the common parameters like start/end row, fq, etc, and then use dynamic props for others.

However if we want to stick with the current approach, I think the description of this property should mention that value is a URL style query string and not just a Lucene query. You also may want to implement a custom validator that ensures this style of query string can be parsed (when EL is not used).